crates/bevy_ecs/src/archetype.rs

```diff
@@ -114,7 +114,10 @@ pub(crate) enum ComponentStatus {
 }
 
 pub(crate) struct AddBundle {
+    /// The target archetype after the bundle is added to the source archetype
     pub archetype_id: ArchetypeId,
+    /// For each component iterated in the same order as the source [`Bundle`](crate::bundle::Bundle),
+    /// indicate if the component is newly added to the target archetype or if it already existed
```

Comment:
I think that's the status on the entity

Reply:
Sorry; of which entity?

From what I understand, the `bundle_status` is computed just by comparing the components of the bundle with the components of the source archetype; no entities are involved.
(https://github.com/cBournhonesque/bevy/blob/7281b0e06c531cac4b9073cbb119b036a2ddfacb/crates/bevy_ecs/src/bundle.rs#L442)

Reply:
Yep @cBournhonesque is right here. The entity's components are based on which archetype it's in, so this comment makes perfect sense.